DATA_OVERLAY.md

```diff
@@ -0,0 +1,57 @@
+## Introduction
+
+In [Metabolic Atlas](https://github.com/MetabolicAtlas/MetabolicAtlas), the visualization of components, such as in the Map Viewer, can be customized using [Data overlay](https://metabolicatlas.org/documentation#Data-overlay).
+
+For each integrated model, Data overlay allows for multiple data types, such as Transcriptomics and Metabolomics. Each data type allows for multiple data sources, as well as the option for the user to upload and use their custom data on the fly. Each data source can contain multiple datasets, which contain values to be used to customize the visualization.
+
+## Data source file requirements
+
+- File format: [`tsv`](https://en.wikipedia.org/wiki/Tab-separated_values)
+- 1st row: This is the header of the file.
+  - 1s column: `id`
+  - 2nd-nth columns: each subsequent column should contain the name of the dataset
+- 2nd-nth rows: each row should contain values that correspond to the header (1st row)
+  - 1st column: component identifier
+  - 2nd-nth columns: a decimal value between 0 and 1
+
+### Example data source file contents
+
+| id              | adipose tissue | adrenal gland |
+| --------------- | -------------- | ------------- |
+| ENSG00000000419 | 0.456          | 0.697         |
+| ENSG00000000938 | 0.291          | 0.57          |
+
+### The 0-1 range
+
+When one dataset is selected in the Data overlay, each value in the 0-1 range corresponds to a color in the following spectrum.
+
+<img width="300" alt="image" src="https://user-images.githubusercontent.com/423498/128190295-e70b6b13-f51e-4076-99ee-3094287466d3.png">
+
+The requirement for all values to be in the 0-1 range is to ensure values from different data sources can be handled the same way by the visualization component.
+
+When adding new data sources, it is important to not excessively limit the range of the values. For example, it might not be good if all of the values are within 0.01 and 0.02, as that would cover a small segment of the color spectrum. Ideally, the minimum value of the data source should be as close to 0 as possible, while the maximum value should be as close to 1 as possible, with as even distribution as possible between for the rest of the values.
```

Comment:
I think this concern should be dealt with separately, and either the paragraph should be updated so it sounds more like a statement (eg "in the case of uneven value distribution, this is what is going to happen"), or completely removed.

Reply:
Can you elaborate on why it should be dealt with separately?

Regarding the evens of distribution, I thought the impact for uneven distribution is implied since the previous sentence mentions that a small total range (0.01 - 0.02) would mean a small segment of the color spectrum is covered. Similarly for uneven distribution, for example if 50% of the data is within 0.01 - 0.02, and 30% is within 0.98 - 0.99, it would have the same problem. Do you think we should be more explicit about this? Also I wanted this to sound more like a recommendation but not a rule, which was the reasoning that this does not sound like a statement.

Reply:
> Can you elaborate on why it should be dealt with separately?

While the documentation is part of supporting multiple data overlay source in `data-files`, the handling of values can/should be updated (dealt with) separately, as we get a better picture of the frontend implementation.
The way I see it, either we fix the problem so we do not need to inform the user, or we tell them clearly there is a pitfall. In this case, it's easier to just be clearer, for now. 

> Do you think we should be more explicit about this?

Yes! Here's what I had in mind:

```suggestion
When adding new data sources, it is important to inspect the range of the values. For example, if all of the values are within 0.01 and 0.02, only a small segment of the color spectrum will be used. Similarly, for uneven distribution, e.g. if 50% of the data is within 0.01 - 0.02 and 30% is within 0.98 - 0.99, the same visualization problem would be encountered.
Ideally, the minimum value of the data source should be as close to 0 as possible, while the maximum value should be as close to 1 as possible, with as even distribution as possible between for the rest of the values.
```

Reply:
Ok, that makes sense. I agree it's better to be clearer for now unless we find out that we could simplify the problem.

Thanks for updating the docs, looks great!